universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java

```diff
@@ -376,7 +376,13 @@ private void putParentArray(Path path) {
     try {
       return this.read(path, Long.class);
     } catch (JsonPathTypeMismatchException e) {
-      return Optional.empty();
+      try {
+        // check if value is a double/float and cast to long if needed
```

Comment:
The Question type definition is to only allow integers, and the FE tries to enforce that, so I think we should have consistency with the backend checking too.

Reply:
This still needs reverting correct?

Reply:
?